src/basic/hash-funcs.h

```diff
@@ -16,11 +16,17 @@ struct hash_ops {
 };
 
 #define _DEFINE_HASH_OPS(uq, name, type, hash_func, compare_func, free_key_func, free_value_func, scope) \
-        _unused_ static void (* UNIQ_T(static_hash_wrapper, uq))(const type *, struct siphash *) = hash_func; \
-        _unused_ static int (* UNIQ_T(static_compare_wrapper, uq))(const type *, const type *) = compare_func; \
+        static void UNIQ_T(static_hash_wrapper, uq)(const void *p, struct siphash *state) { \
```

Comment:
can you add a comment somewhere close that explains why we do this? I have the suspicion people who read this will wonder why do this wrapping here...

just a quick note that LLVM version so-and-so with option so-and-so is strict regarding function casts.